packages/std/src/ado_contract/permissioning.rs

```diff
@@ -79,36 +81,54 @@ impl<'a> ADOContract<'a> {
         let action_string: String = action.into();
         let actor_string: String = actor.into();
 
-        if self.is_contract_owner(store, actor_string.as_str())? {
+        if self.is_contract_owner(deps.as_ref().storage, actor_string.as_str())? {
             return Ok(());
         }
 
-        let permission = Self::get_permission(store, action_string.clone(), actor_string.clone())?;
+        let permission = Self::get_permission(
+            deps.as_ref().storage,
+            action_string.clone(),
+            actor_string.clone(),
+        )?;
         let permissioned_action = self
             .permissioned_actions
-            .may_load(store, action_string.clone())?
+            .may_load(deps.storage, action_string.clone())?
             .unwrap_or(false);
         match permission {
-            Some(mut permission) => {
-                ensure!(
-                    permission.is_permissioned(&env, permissioned_action),
-                    ContractError::Unauthorized {}
-                );
-
-                // Consume a use for a limited permission
-                if let Permission::Limited { .. } = permission {
-                    permission.consume_use()?;
-                    permissions().save(
-                        store,
-                        (action_string.clone() + actor_string.as_str()).as_str(),
-                        &PermissionInfo {
-                            action: action_string,
-                            actor: actor_string,
-                            permission,
-                        },
-                    )?;
-                }
-
+            Some(mut some_permission) => {
+                match some_permission {
+                    Permission::Local(ref mut local_permission) => {
+                        ensure!(
+                            local_permission.is_permissioned(&env, permissioned_action),
+                            ContractError::Unauthorized {}
+                        );
+
+                        // Consume a use for a limited permission
+                        if let LocalPermission::Limited { .. } = local_permission {
+                            local_permission.consume_use()?;
+                            permissions().save(
+                                deps.storage,
+                                (action_string.clone() + actor_string.as_str()).as_str(),
+                                &PermissionInfo {
+                                    action: action_string,
+                                    actor: actor_string,
+                                    permission: some_permission,
+                                },
+                            )?;
+                        }
+                    }
+                    Permission::Contract(address_list) => {
```

Comment:
variable name `address_list` seems to be somewhat confusing here. It can be analyzed as a list of addresses or address of the `address_list` ado. I suggest using more general variable name as we have no guarantee that only `address_list` will be used for permission system.

Reply:
named it `contract_address`. 
Same for below.